### PR TITLE
[SHELL32] Improve SHBrowseForFolder

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-7592](https://jira.reactos.org/browse/CORE-7592)

AFTER:
after

- Implement `VK_DELETE`.
- Use `LPSTR_TEXTCALLBACKW` in `LV_RenameItem`.

### CODE REVIEW

File: dll/win32/shell32/wine/brsfolder.c [hunk 1]
```diff
@@ -625,6 +628,30 @@ static HRESULT BrsFolder_Rename(browse_info *info, HTREEITEM rename)
     return S_OK;
 }
 
+#ifdef __REACTOS__
+static void
+BrsFolder_Treeview_DeleteBang(browse_info *info, HTREEITEM selected_item)
```
Comment (learn-more) [inline dll/win32/shell32/wine/brsfolder.c:633]: `DeleteBang`?

Comment (katahiromz) [reply dll/win32/shell32/wine/brsfolder.c:633]: Is there the better name?

Comment (katahiromz) [reply dll/win32/shell32/wine/brsfolder.c]: `s/DeleteBang/DeleteFolder/`

**Comment by katahiromz** [comment]
There is a bug that the pre-created folder won't be deleted visually.

File: dll/win32/shell32/wine/brsfolder.c [hunk 2]
```diff
@@ -1104,16 +1127,23 @@ static HTREEITEM BrsFolder_FindItemByPidl(browse_info *info, LPCITEMIDLIST pidl,
...
+#ifndef __REACTOS__
     HRESULT hr;
+#endif
 
     item.mask = TVIF_HANDLE | TVIF_PARAM;
     item.hItem = hItem;
     SendMessageW(info->hwndTreeView, TVM_GETITEMW, 0, (LPARAM)&item);
     item_data = (TV_ITEMDATA *)item.lParam;
 
+#ifdef __REACTOS__
```
Comment (yagoulas) [inline dll/win32/shell32/wine/brsfolder.c:1139]: why is this needed?

Comment (katahiromz) [reply dll/win32/shell32/wine/brsfolder.c:1139]: This source code is based on Wine source. For comparison to Wine source, I inserted `#ifdef __REACTOS__` ... `#endif`.

**Comment by Mouvedia** [comment]
"browse for folder" and "make new folder" sound weird to me.
Would Microsoft be able to protect their button text labels? That's paranoid.

File: dll/win32/shell32/wine/brsfolder.c (hunk 2 above)
Comment (katahiromz) [reply dll/win32/shell32/wine/brsfolder.c:1139]: `item_data->lpifq` and `pidl` are absolute PIDLs. We cannot compare them with `IShellFolder_CompareIDs`.

**Comment by katahiromz** [comment]
Please wait for #2659.

**Comment by katahiromz** [comment]
The bug is fixed. Ready to go!

**Comment by katahiromz** [comment]
Splitted into #2660 and #2661. Closed.
